test/unit/runtime-rules-hotloading.test.ts

```diff
@@ -0,0 +1,92 @@
+import { readFileSync } from 'fs';
+import * as path from 'path';
+import * as loadFilterRules from '../../lib/filter-rules-loading';
+
+function loadFixture(name: string) {
+  const fixturePath = path.join(__dirname, '..', 'fixtures', name);
+  const fixture = readFileSync(fixturePath, { encoding: 'utf-8' });
+
+  return fixture;
+}
+
+describe('filter Rules Loading', () => {
+  it('Loads normal accept file', () => {
+    const rules = JSON.parse(loadFixture(path.join('accept', 'ghe.json')));
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
+  });
+
+  it('Skip injection if no or invalid IAC extensions', () => {
+    process.env.ACCEPT_IAC = 'rf';
+    const rules = JSON.parse(loadFixture(path.join('accept', 'ghe.json')));
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
+  });
+
+  it('Injection of valid IAC extensions', () => {
+    process.env.ACCEPT_IAC = 'tf,yaml, json,yml,tpl';
+    const rules = JSON.parse(
+      loadFixture(path.join('accept', 'ghe-with-iac-rules.json')),
+    );
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
+    delete process.env.ACCEPT_IAC;
+  });
+
+  it('Injection of valid IAC extensions', () => {
+    process.env.ACCEPT_IAC = 'tf,json';
+    const rules = JSON.parse(
+      loadFixture(path.join('accept', 'ghe-with-iac-rules-tf-json-only.json')),
+    );
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
+    delete process.env.ACCEPT_IAC;
+  });
+
+  it('Injection of valid CODE rules', () => {
+    process.env.ACCEPT_CODE = 'true';
+
+    const rules = JSON.parse(
+      loadFixture(path.join('accept', 'ghe-with-code-rules.json')),
+    );
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
+    delete process.env.ACCEPT_CODE;
+  });
+
+  it('Injection of valid CODE rules and IAC extensions (yaml only)', () => {
+    process.env.ACCEPT_CODE = 'true';
+    process.env.ACCEPT_IAC = 'yaml';
+    const rules = JSON.parse(
+      loadFixture(path.join('accept', 'ghe-with-code-rules-and-iac-yaml.json')),
+    );
+    const loadedRules = loadFilterRules(
+      'ghe.json',
+      path.join(__dirname, '..', 'fixtures/accept'),
+    );
+
+    expect(loadedRules).toEqual(rules);
```

Comment:
nit: rather than having to create these custom fixtures, you could use Jest [snapshots](https://jestjs.io/docs/snapshot-testing). They would save the results and compare them between test making it potentially easier to spot regressions.

Reply:
done